packages/loopring_v3/contracts/core/impl/libexchange/ExchangeBlocks.sol

```diff
@@ -249,6 +245,12 @@ library ExchangeBlocks
                 );
                 uint txDataOffset = 0;
 
+                if (approved &&
+                    txType != ExchangeData.TransactionType.WITHDRAWAL &&
+                    txType != ExchangeData.TransactionType.DEPOSIT) {
+                    continue;
```

Comment:
The `approved` system was only used for AMM pool contracts (contracts we fully control), which don't do any deposits/withdrawals so the behavior didn't matter. Now we also use it for the Bridge, and the Bridge needs to approve withdrawals (but only from its own address), so now the behavior does matter for withdrawals a bit (but things like `withdrawalRecipient` still don't really matter because the Bridge is a contract so no practical way we'd use that without EdDSA signatures).

There are 2 options:
1. We skip all checks for deposits/withdrawals in the exchange (previous behaviour). This is possible, because e.g. the bridge can just transfer funds to the deposit contract and then just do the deposit transaction to reflect that. The same with withdrawals, the bridge could in theory have direct access to funds in the deposit contract and the withdrawal transaction again is just done to reflect that change.
2. We still do all normal checks for deposits/withdrawals (current behaviour).

I have opted for 2) because with 1) the security of the complete rollup depends on the security of the bridge (even though it could be more efficient to do 1).